Extensions/Manage/Dnn.PersonaBar.Users/Components/RegisterController.cs

```diff
@@ -103,7 +104,7 @@ public UserBasicDto Register(RegisterationDetails registerationDetails)
             //ensure this user doesn't exist
             if (!string.IsNullOrEmpty(username) && UserController.GetUserByName(portalSettings.PortalId, username) != null)
             {
-                throw new Exception(Localization.GetString("RegistrationUsernameAlreadyPresent",
+                throw new SecurityException(Localization.GetString("RegistrationUsernameAlreadyPresent",
```

Comment:
Let's not use `SecurityException`, let's create a new exception type, for instance `BadRequestException` and throw that one instead.

Reply:
> We cannot just fix this particular validation and leave all other validations as is. There are some other validations that will still return a 500 error. Rule should be:
> 
> if the request cannot be processed because user supplied wrong arguments (this includes user already exists), then throw a BadRequest exception and then return a 400 HTTP response.
> in case of any unhandled exceptions, return a 500 response
> 
> Also please make sure that Prompt commands still work

Currently Prompt Command for existing username is not throwing any http error in browser console. i.e. 400

Reply:
- New exception type added 
- All other exceptions are already return with 500 response
- Prompt Command for 'new-user' is not returning any http error response for existing user

Reply:
I see 9 more cases to fix where we are throwing `Exception` and we should be throwing `BadRequestException`

Reply:
@Mhtshum ^^